cmd/server-mux.go

```diff
@@ -194,6 +199,9 @@ func newListenerMux(listener net.Listener, config *tls.Config) *ListenerMux {
 				return
 			}
 
+			// Enable Read timeout
+			conn.SetReadDeadline(time.Now().Add(defaultTcpReadTimeout))
+
 			// Enable keep alive for each connection.
```

Comment:
What we need is doing "conn" list ourselves i.e basically make sure that we don't run out of RLimit.Max and handle it ourselves.. So that we can reject new connection attempts gracefully.